packages/core/src/utils/iterators.ts

```diff
@@ -81,3 +81,34 @@ export function join<T>(iterable: Iterable<T>, glue: string): string {
 
   return result;
 }
+
+/**
+ * Like {@link Array#filter}, but works with any iterable.
+ *
+ * @param iterable
+ * @param cb
+ */
+export function *filter<In>(
+  iterable: Iterable<In>,
+  cb: (item: In) => boolean,
+): Generator<In, void> {
+  for (const item of iterable) {
+    if (cb(item)) {
+      yield item;
+    }
+  }
+}
+
+/**
+ * Function to get the number of items in an iterable.
+ *
+ * @param iterable
+ */
+export function size(iterable: Iterable<unknown>): number {
```

Comment:
the only use of this function is on an array, which would can use `.length` to be more efficient

Unless you meant to use it on `attributes` instead of on `returnFields`, but in that case I recommend creating a utility function called `isEmptyIterable`, which would only check the first value

It would simply be:

```ts
export default function isEmptyIterable(iterable: Iterable<unknown>): number {
	for (const ignore of iterable) {
		return false;
	}

	return true;
}
```

Reply:
After rebuilding the returning internal functions, this is no longer required.